.golangci.yml

```diff
@@ -55,14 +22,47 @@ linters:
     - predeclared
     - revive
     - staticcheck
-    - stylecheck
     - thelper
     - tparallel
-    - typecheck
     - unconvert
     - unparam
     - whitespace
     - wsl
+  settings:
+    errcheck:
+      check-type-assertions: true
+    exhaustive:
+      default-signifies-exhaustive: true
+    goconst:
+      min-len: 2
+      min-occurrences: 3
+    gocritic:
+      enabled-tags:
+        - diagnostic
+        - experimental
+        - opinionated
+        - performance
+        - style
+    govet:
+      enable-all: true
+    nolintlint:
+      require-explanation: true
+      require-specific: true
+    wsl:
+      allow-assign-and-anything: true
+      allow-cuddle-declarations: true
+      allow-assign-and-call: true
+    depguard:
+      rules:
+        main:
+          list-mode: lax
+          allow:
+            - "github.com/uber/h3-go/v4"
+
+formatters:
+  enable:
+    - gofmt
+    - goimports
```

Comment:
i didn't vet all these settings, any changes of note besides the comments being correctly enforced and the copyloop linter getting enabled?